cmd/tailscale/cli/file.go

```diff
@@ -387,54 +389,45 @@ func receiveFile(ctx context.Context, wf apitype.WaitingFile, dir string) (targe
 	if err != nil {
 		return "", 0, fmt.Errorf("opening inbox file %q: %w", wf.Name, err)
 	}
+	defer rc.Close()
 	f, err := openFileOrSubstitute(dir, wf.Name, getArgs.conflict)
 	if err != nil {
 		return "", 0, err
 	}
 	_, err = io.Copy(f, rc)
-	rc.Close()
 	if err != nil {
+		f.Close()
 		return "", 0, fmt.Errorf("failed to write %v: %v", f.Name(), err)
 	}
 	return f.Name(), size, f.Close()
 }
 
-func runFileGet(ctx context.Context, args []string) error {
-	if len(args) != 1 {
-		return errors.New("usage: file get <target-directory>")
-	}
-	log.SetFlags(0)
-
-	dir := args[0]
-	if dir == "/dev/null" {
-		return wipeInbox(ctx)
-	}
-
-	if fi, err := os.Stat(dir); err != nil || !fi.IsDir() {
-		return fmt.Errorf("%q is not a directory", dir)
-	}
-
+func runFileGetOneBatch(ctx context.Context, dir string) []error {
 	var wfs []apitype.WaitingFile
 	var err error
-	for {
+	var errs []error
+	for len(errs) < 2 {
 		wfs, err = tailscale.WaitingFiles(ctx)
 		if err != nil {
-			return fmt.Errorf("getting WaitingFiles: %w", err)
+			errs = append(errs, fmt.Errorf("getting WaitingFiles: %w", err))
 		}
-		if len(wfs) != 0 || !getArgs.wait {
+		if len(wfs) != 0 || !(getArgs.wait || getArgs.loop) {
 			break
 		}
 		if getArgs.verbose {
 			printf("waiting for file...")
 		}
 		if err := waitForFile(ctx); err != nil {
-			return err
+			errs = append(errs, err)
 		}
 	}
 
-	var errs []error
 	deleted := 0
 	for _, wf := range wfs {
+		if len(errs) > 100 {
+			// Likely, everything is broken. Let's just stop.
```

Comment:
just stop this batch, right? the outer loop will still run forever?

Reply:
that's right, the outer loop will still run forever.  this short-circuiting is mostly to help limit logging (or error memory map size as we briefly had in this cl) if we get flooded with incoming files and have persistent errors.